cli/ts/genProofs.ts

```diff
@@ -336,23 +360,34 @@ const genProofs = async (args: any) => {
         return 
     }
 
+    let maciState: MaciState
     // Build an off-chain representation of the MACI contract using data in the contract storage
-
-    // some rpc endpoint like bsc chain has limitation to retreive history logs
-    let fromBlock = 0
-    const txHash = args.transaction_hash
-    if (txHash) {
-        const txn = await signer.provider.getTransaction(txHash);
-        fromBlock = txn.blockNumber
+    if (args.state_file) {
+        // @todo actually read the file first 
+        const content = JSON.parse(fs.readFileSync(args.state_file, 'utf-8').toString())
+        maciState = MaciState.fromJSON(content)
+        // ensure we merge all messages
+        maciState.polls.forEach((poll: Poll) => poll.mergeAllMessages()) 
+    } else {
```

Comment:
This change introduce to reading from a state file if `args.state_file` is provided, and then generating the proof based on the JSON content. However, there appears to be no test coverage to verify the correctness of the serialization/deserialization process and the subsequent proof generation using the parsed data. Could we introduce unit test that validate the correct functioning of the proof generation when using parsed data?